examples/creatives/hover_nonlinear.js

```diff
@@ -0,0 +1,143 @@
+        super.onStart(eventData);
+        this.addActions_();
+    }
+
+    /**
+     * Adds actions to different buttons available on the overlay.
+     * @private
+     */
+    addActions_() {
+        this.sendMessageOnClick_('close_ad', CreativeMessage.REQUEST_STOP);
+        this.onHover_('content_container', 'mouseover');
+        this.onMouseOut_('content_container', 'mouseout');
+    }
+
+    /**
+     * Sends a SIMID message whenever an element is clicked.
+     * @param {String} elementName The name of the element.
+     * @param {String} message The message to send to the player.
+     * @private
+     */
+    sendMessageOnClick_(elementName, message) {
+        const sendMessageFunction = () => {this.simidProtocol.sendMessage(message);}
+        document.getElementById(elementName).addEventListener('click', sendMessageFunction);
+    }
+
+    /**
+     * Adds a hover event listener to the contents of the iframe that expands the iframe.
+     * @param {String} elementName The name of the element.
+     * @param {Event} event The event performed on the element.
+     * @private
+     */   
+    onHover_(elementName, event) {
+        const expandOnHoverFunction = () => {
+            const newDimensions = {};
+            let resizeParams = {};
+
+            //if ad is at the top quarter of the player
+            if (this.initialDimensions_.y < (this.videoDimensionHeight * QUARTER_THE_SIZE)) {
+                newDimensions.height = (this.videoDimensionHeight * THIRTY_FIVE_PERCENT);
+                newDimensions.x = 0;
+                newDimensions.y = this.initialDimensions_.y;
+                newDimensions.width = this.videoDimensionWidth;
+
+                resizeParams = {
+                    creativeDimensions: newDimensions,
+                    videoDimensions: this.environmentData.videoDimensions,
+                };
+            
+                this.requestResize(resizeParams);
+            }
+
+            //if ad is between the top quarter and half of the player
+            else if (this.initialDimensions_.y > (this.videoDimensionHeight * QUARTER_THE_SIZE) && 
+                this.initialDimensions_.y < (this.videoDimensionHeight * HALF_THE_SIZE)) {
+                    newDimensions.height = (this.videoDimensionHeight * THIRTY_FIVE_PERCENT);
+                    newDimensions.x = 0;
+                    newDimensions.y = this.videoDimensionHeight * QUARTER_THE_SIZE;
+                    newDimensions.width = this.videoDimensionWidth;
+
+                    resizeParams = {
+                        creativeDimensions: newDimensions,
+                        videoDimensions: this.environmentData.videoDimensions,
+                    };
+                
+                    this.requestResize(resizeParams);
+            }
+
+            //if ad is between the top half and three quarters of the player
+            else if (this.initialDimensions_.y > (this.videoDimensionHeight * HALF_THE_SIZE) && 
```

Comment:
It feels a little more complex than needed.  I'll have to see how this works in practice.

In general for this type of problem, lookin at where the nonlinear is, might be a wrong path, instead ideally just focus on how much you want it to expand.

That said, I don't think there's a bunch of time for fixing, since the internship is coming to a close.  If you do have more time to make this simpler though, have a try at it.

Reply:
I drastically improved the logic, take a look at new commits